src/orders/components/OrderTransactionRefundPage/components/OrderTransactionRefundTable/OrderTransactionRefundTable.tsx

```diff
@@ -0,0 +1,196 @@
+  field: string;
+  lineId: string;
+}
+
+export const OrderTransactionRefundTable: React.FC<OrderTransactionRefundTableProps> = ({
+  order,
+  draftRefund,
+  control,
+  onChange,
+  onEditReasonModal,
+  linesToRefund,
+  refundFields,
+  refundFieldsUpdate,
+}) => {
+  const locale = useLocale();
+
+  return (
+    <GridTable height="100%" paddingX={6}>
+      <GridTable.Colgroup>
+        <GridTable.Col __width="45%" />
+        <GridTable.Col __width="20%" />
+        <GridTable.Col __width="25%" />
+        <GridTable.Col __width="10%" />
+      </GridTable.Colgroup>
+      <GridTable.Body>
+        {refundFields?.map((field, ix) => {
+          const line = order?.lines[ix];
+
+          if (!line) {
+            return;
+          }
+
+          const maxQtyToRefund = getMaxQtyToRefund({
+            rowData: { id: line.id, quantity: line.quantity },
+            order,
+            draftRefund,
+          });
+
+          return (
+            <GridTable.Row key={field.id}>
+              <GridTable.Cell>
+                <Box
+                  display="grid"
+                  __gridTemplateColumns="min-content auto"
+                  gap={2}
+                  alignItems="start"
+                >
+                  <Box width={8} height={8}>
+                    <img src={line.thumbnail?.url} />
+                  </Box>
+                  <Box overflow="hidden" minWidth={0}>
+                    <Text size={2} textOverflow="ellipsis" overflow="hidden" whiteSpace="nowrap">
+                      {line.productName}
+                    </Text>
+                    <Box overflow="hidden" textOverflow="ellipsis" whiteSpace="nowrap">
+                      <Text size={2} color="default2" textOverflow="ellipsis" overflow="hidden">
+                        {line.variantName}
+                      </Text>
+                    </Box>
+                  </Box>
+                </Box>
+              </GridTable.Cell>
+              <GridTable.Cell>
+                <Box
+                  display="flex"
+                  flexDirection="column"
+                  overflow="hidden"
+                  flexShrink="1"
+                  __minWidth="0px"
+                >
+                  {`${line.quantity} ⨉ `}
+                  {formatMoney(line.unitPrice.gross, locale.locale)}
+                  {maxQtyToRefund !== line.quantity && (
+                    <Text size={2} whiteSpace="nowrap" color="default2">
+                      {maxQtyToRefund === 0 ? (
+                        <FormattedMessage {...refundTableMessages.allItemsRefunded} />
+                      ) : (
+                        <FormattedMessage
+                          {...refundTableMessages.alreadyRefunded}
+                          values={{
+                            value: (line.quantity - maxQtyToRefund).toString(),
+                          }}
+                        />
+                      )}
+                    </Text>
+                  )}
```

Comment:
**suggestion:** You put many if statements here and some logic - make a component of it that presents the current state of a given line